File: spec/python/test_optional_id.py
import unittest

from optional_id import OptionalId

class TestOptionalId(unittest.TestCase):
    def test_optional_id(self):
        r = OptionalId.from_file("src/fixed_struct.bin")

        self.assertEqual(r._unnamed0, 0x50)
        self.assertEqual(r._unnamed1, 0x41)
        self.assertEqual(r._unnamed2, b"\x43\x4b\x2d\x31\xff")

File: spec/lua/test_optional_id.lua
local luaunit = require("luaunit")

require("optional_id")

TestOptionalId = {}

function TestOptionalId:test_optional_id()
    local r = OptionalId:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r._unnamed0, 80)
    luaunit.assertEquals(r._unnamed1, 65)
    luaunit.assertEquals(r._unnamed2, "\067\075\045\049\255")
end
